File: pkg/models/agent.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Agent represents a named AI agent with scoped query access.
type Agent struct {
	ID              uuid.UUID  `json:"id"`
	ProjectID       uuid.UUID  `json:"project_id"`
	Name            string     `json:"name"`
	APIKeyEncrypted string     `json:"-"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
	LastAccessAt    *time.Time `json:"last_access_at"`
	MCPCallCount    int64      `json:"mcp_call_count"`
}

File: ui/src/types/agent.ts
export interface Agent {
  id: string;
  name: string;
  query_ids: string[];
  created_at: string;
  updated_at?: string;
  last_access_at?: string | null;
  mcp_call_count: number;
}

export interface AgentCreateResponse extends Agent {
  api_key: string;
}

export interface AgentListResponse {
  agents: Agent[];
}

export interface AgentKeyResponse {
  key: string;
  masked: boolean;
}
